### PR TITLE
[MB-9026] add components to  the review shipment weights sidebar

### PR DESCRIPTION
**Pull request by pearl-truss**
## Description

Incorporates the review billable weight content into the document viewer sidebar in the review billable weights flow. This PR also added in some logic for:
- displaying the correct flags in the ship container component for missing estimated or reweighs
- hiding the reweigh related fields on the shipment card when a reweigh hasn't been requested on a shipment and it 
- the max billable weight exceeded flag to the edit max billable weight sidebar \
- the warning flags on the shipment sidebar for the missing info flag (appears when a reweigh was requested and the shipment doesnt have a reweigh weight or when the shipment is missing an estimated weight) 
- the over weight shipment flag appears when the shipment's original weight is more than 100% of the estimated weight

## Reviewer Notes

- This should be merged after https://github.com/transcom/mymove/pull/7359 
- Double check that the logic is correct

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make server_run
make office_client_run
```
Log into TIO account
Click on any shipment any of these locators:

BILWEI
MULTRW
MISHRW
MAXCED

View payment requests
Click on Review weights
Expected result: Max billable weights page should have relevant information inside of components. You should be able to click the Review Shipments button in the sidebar and see the shipment specific weight info. If there's multiple shimpents, clicking Next Shipment will display the next shipment 

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-9026) for this change
* [specs for the page](https://app.zeplin.io/project/6112d5999a811689179723b4/screen/611ac58296d7a833928b98b9) 

## Screenshots

Screen Shot 2021-09-14 at 3 45 42 PM
Screen Shot 2021-09-14 at 3 45 48 PM
Screen Shot 2021-09-14 at 3 46 09 PM

### CODE REVIEW

File: src/utils/shipmentWeights.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// If the shipment's billable weight cap is greater than 110% of the estimated weight,
+// then the shipment is overweight
+export function shipmentIsOverweight(estimatedWeight, weightCap) {
+  return weightCap > estimatedWeight * 1.1;
+}
+
+export function calcWeightRequested(mtoShipments) {
```
Comment (YanZ777) [inline src/utils/shipmentWeights.js:7]: There are custom hooks that you can use inside of `custom.js`!

**Review by YanZ777 — CHANGES_REQUESTED** [review]

File: src/pages/Office/ReviewBillableWeight/ReviewBillableWeight.jsx [hunk 2]
```diff
@@ -49,10 +104,62 @@ export default function ReviewBillableWeight() {
...
+          >
             <DocumentViewerSidebar.Content>
-              Review shipment weight content should go in here
+              <div className={reviewBillableWeightStyles.contentContainer}>
+                {((!selectedShipment.reweigh?.weight && selectedShipment.reweigh?.requestedAt) ||
+                  !selectedShipment.primeEstimatedWeight) && (
+                  <Alert slim type="warning">
+                    Shipment missing information
+                  </Alert>
+                )}
```
Comment (donaldthai) [inline src/pages/Office/ReviewBillableWeight/ReviewBillableWeight.jsx:120]: I think we also need to include the case of missing estimated weight. https://app.zeplin.io/project/6112d5999a811689179723b4/screen/611ac58296d7a833928b98b9

Comment (pearl-truss) [reply src/pages/Office/ReviewBillableWeight/ReviewBillableWeight.jsx:120]: The conditional statement here includes `!selectedShipment.primeEstimatedWeight`. Or are you referring to something else?

File: src/pages/Office/ReviewBillableWeight/ReviewBillableWeight.jsx [hunk 3]
```diff
@@ -36,7 +74,24 @@ export default function ReviewBillableWeight() {
         {sidebarType === 'MAX' ? (
           <DocumentViewerSidebar title="Review weights" subtitle="Edit max billable weight" onClose={handleClose}>
             <DocumentViewerSidebar.Content>
-              Review max billable weight content should go in here
+              {maxBillableWeight > weightAllowance && (
+                <Alert slim type="error">
+                  {`Max billable weight exceeded. \nPlease resolve.`}
+                </Alert>
+              )}
+              <WeightSummary
```
[the remark this answers is not in the document]
Comment (pearl-truss) [reply src/pages/Office/ReviewBillableWeight/ReviewBillableWeight.jsx]: this is addressed

File: src/utils/shipmentWeights.js (hunk 1 above)
Comment (pearl-truss) [reply src/utils/shipmentWeights.js]: thanks! this is addressed

File: src/pages/Office/ReviewBillableWeight/ReviewBillableWeight.jsx (hunk 2 above)
Comment (donaldthai) [reply src/pages/Office/ReviewBillableWeight/ReviewBillableWeight.jsx:122]: Oops, didn't see that part! Looks goooddd!

**Review by donaldthai — APPROVED** [review]
🚀

**Review by cosburn — APPROVED** [review]
Reviewed screenshot of requested style changes, looks good!

(slack thread for historical reference: https://ustcdp3.slack.com/archives/C0292K7MY31/p1631725632135000?thread_ts=1631719857.130700&cid=C0292K7MY31)

**Review by YanZ777 — APPROVED** [review]
Looks good to me! Thanks for making those changes.

**Pull request merged**